mysql/mysqldump.go

```diff
@@ -0,0 +1,578 @@
+	}
+	return arraybound
+}
+
+func processInsertStmt(conv *internal.Conv, stmt *ast.InsertStmt) {
+	if stmt.Table == nil {
+		logStmtError(conv, stmt, fmt.Errorf("source table is nil"))
+		return
+	}
+	srcTable, err := getTableNameInsert(stmt.Table)
+	if err != nil {
+		logStmtError(conv, stmt, fmt.Errorf("can't get source table name: %w", err))
+		return
+	}
+	if conv.SchemaMode() {
+		conv.Stats.Rows[srcTable] += int64(len(stmt.Lists))
+		return
+	}
+	spTable, err1 := internal.GetSpannerTable(conv, srcTable)
+	if err1 != nil {
+		logStmtError(conv, stmt, fmt.Errorf("can't get spanner table name for source table '%s' : err=%w", srcTable, err1))
+		return
+	}
+	spSchema, ok1 := conv.SpSchema[spTable]
+	srcSchema, ok2 := conv.SrcSchema[srcTable]
+	if !ok1 || !ok2 {
+		conv.Unexpected(fmt.Sprintf("Can't get schemas for table %s", srcTable))
+		conv.Stats.BadRows[srcTable] += conv.Stats.Rows[srcTable]
+		return
+	}
+	srcCols, err2 := getCols(stmt)
+	if err2 != nil {
+		// In MySQL, column names might not be specified in insert statement so instead of
+		// throwing error we will try to retrieve columns from source schema
+		srcCols = conv.SrcSchema[srcTable].ColNames
+		if len(srcCols) == 0 {
+			conv.Unexpected(fmt.Sprintf("Can't get columns for table %s", srcTable))
+			conv.Stats.BadRows[srcTable] += conv.Stats.Rows[srcTable]
+			return
+		}
+	}
+	spCols, err3 := internal.GetSpannerCols(conv, srcTable, srcCols)
+	if err3 != nil {
+		conv.Unexpected(fmt.Sprintf("Can't get spanner columnss for table %s: err=%s", srcTable, err3))
+		conv.Stats.BadRows[srcTable] += conv.Stats.Rows[srcTable]
+		return
+	}
+	var values []string
+	if stmt.Lists == nil {
+		logStmtError(conv, stmt, fmt.Errorf("Can't get column values"))
+		return
+	}
+	for _, row := range stmt.Lists {
+		values, err = getVals(row)
+		ProcessDataRow(conv, srcTable, srcCols, srcSchema, spTable, spCols, spSchema, values)
+	}
+}
+
+func getCols(stmt *ast.InsertStmt) ([]string, error) {
+	if stmt.Columns == nil {
+		return nil, fmt.Errorf("No columns found in insert statement ")
+	}
+	var colnames []string
+	for _, column := range stmt.Columns {
+		colnames = append(colnames, column.OrigColName())
+	}
+	return colnames, nil
+}
+
+func getVals(row []ast.ExprNode) ([]string, error) {
+	if len(row) == 0 {
+		return nil, fmt.Errorf("Found row with zero length")
+	}
+	var values []string
+	for _, item := range row {
+
+		switch valueNode := item.(type) {
+		case *driver.ValueExpr:
+			values = append(values, fmt.Sprintf("%v", valueNode.GetValue()))
+		default:
+			return nil, fmt.Errorf("unexpected value node %T", valueNode)
+		}
+	}
+	return values, nil
+}
+
+func getTableNameInsert(stmt *ast.TableRefsClause) (string, error) {
+	if stmt.TableRefs == nil {
+		return "", fmt.Errorf("tablerefs is empty, can't build table name")
+	}
+
+	if stmt.TableRefs.Left == nil {
+		return "", fmt.Errorf("tablerefs.Left is empty, can't build table name")
+	}
+
+	switch table := stmt.TableRefs.Left.(type) {
```

Comment:
I think we should use `if` instead of `switch`: 

```go
if s, ok := stmt.TableRefs.Left.(*ast.TableSource); ok {

} else {

}
```

Reply:
https://github.com/cloudspannerecosystem/harbourbridge/blob/fa3d015562a86e51157a983f900b07a1fb5ca62d/mysql/mysqldump.go#L552-L560
Done